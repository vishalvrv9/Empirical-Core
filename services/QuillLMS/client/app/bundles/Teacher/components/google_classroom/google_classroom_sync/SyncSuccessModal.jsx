import React from 'react'
import $ from 'jquery'
import pluralize from 'pluralize'

import Modal from 'react-bootstrap/lib/Modal';

export default React.createClass({
    render: function() {
      let syncedCount;
      if (this.props.data && this.props.data.selectedClassrooms) {
       syncedCount = this.props.data.selectedClassrooms.length;
     } else {
       syncedCount = 0;
     }
        return (
            <Modal {...this.props} show={this.props.show} onHide={this.props.hideModal} dialogClassName='google-classroom-modal sync-success' restoreFocus={true}>
              <Modal.Body>
                  <img className='pull-right react-bootstrap-close' onClick={this.props.hideModal} src='/images/close_x.svg' alt='close-modal'/>
                  <h1 className='q-h2'><i className="fa fa-check" aria-hidden="true"></i>Great! You've synced {`${syncedCount} ${pluralize('class', syncedCount)}`}</h1>
                  <h3 className='subheader'>It may take up to <strong>5 minutes</strong> to import your students.</h3>
                  <p>
                    Please have your students log in using their Google accounts. If you have any questions, you can check <a href='www.support.quill.org'>support.quill.org</a>
                  </p>
                    <a className="q-button cta-button bg-quillgreen text-white back-to-profile" href='/'>
                      Got It!
                    </a>
              </Modal.Body>
            </Modal>
        )
    }

})
