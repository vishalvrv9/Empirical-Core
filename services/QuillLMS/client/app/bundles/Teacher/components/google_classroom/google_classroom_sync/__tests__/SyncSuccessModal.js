import React from 'react';
import { shallow } from 'enzyme';
import SyncSuccessModal from '../SyncSuccessModal.jsx';

describe('the SyncSuccessModal component', () => {
  const wrapper = shallow(<SyncSuccessModal data={{ selectedClassrooms: [1, 2, 3, 4, 5], }} />);

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should show the number of classes being synced', () => {
    expect(wrapper.find('h1').first().text()).toBe('Great! You\'ve synced 5 classes');
  });
});
