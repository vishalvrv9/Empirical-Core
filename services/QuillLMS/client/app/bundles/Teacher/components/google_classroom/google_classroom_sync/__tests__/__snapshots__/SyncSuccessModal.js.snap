// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the SyncSuccessModal component should render 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  data={
    Object {
      "selectedClassrooms": Array [
        1,
        2,
        3,
        4,
        5,
      ],
    }
  }
  dialogClassName="google-classroom-modal sync-success"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "modals": Array [],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={false}
>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <img
      alt="close-modal"
      className="pull-right react-bootstrap-close"
      src="/images/close_x.svg"
    />
    <h1
      className="q-h2"
    >
      <i
        aria-hidden="true"
        className="fa fa-check"
      />
      Great! You've synced 
      5 classes
    </h1>
    <h3
      className="subheader"
    >
      It may take up to 
      <strong>
        5 minutes
      </strong>
       to import your students.
    </h3>
    <p>
      Please have your students log in using their Google accounts. If you have any questions, you can check 
      <a
        href="www.support.quill.org"
      >
        support.quill.org
      </a>
    </p>
    <a
      className="q-button cta-button bg-quillgreen text-white back-to-profile"
      href="/"
    >
      Got It!
    </a>
  </ModalBody>
</Modal>
`;
